driver-core/src/test/java/com/datastax/driver/core/SchemaChangesTest.java

```diff
@@ -68,81 +99,132 @@ public void teardown() {
         return new Object[][]{ { "lowercase" }, { "\"CaseSensitive\"" } };
     }
 
-    @Test(groups = "short", dataProvider = "existingKeyspaceName")
-    public void should_notify_of_table_creation(String keyspace) {
-        session.execute(String.format("CREATE TABLE %s.table1(i int primary key)", keyspace));
+    @DataProvider(name = "newKeyspaceName")
+    public static Object[][] newKeyspaceName() {
+        return new Object[][]{ { "lowercase2" }, { "\"CaseSensitive2\"" } };
+    }
 
-        for (Metadata m : metadatas)
-            assertThat(m.getKeyspace(keyspace).getTable("table1"))
-                .isNotNull();
+    @AfterMethod
+    public void resetListeners(){
+        for (SchemaChangeTestListener listener : listeners) {
+            listener.reset(); // reset event counter
+            reset(listener); // reset Mockito spy
+        }
     }
 
     @Test(groups = "short", dataProvider = "existingKeyspaceName")
-    public void should_notify_of_table_update(String keyspace) {
-        session.execute(String.format("CREATE TABLE %s.table1(i int primary key)", keyspace));
-        session.execute(String.format("ALTER TABLE %s.table1 ADD j int", keyspace));
-
+    public void should_notify_of_table_creation(String keyspace) throws InterruptedException {
+        execute(CREATE_TABLE, keyspace);
         for (Metadata m : metadatas)
-            assertThat(m.getKeyspace(keyspace).getTable("table1").getColumn("j"))
-                .isNotNull();
+            assertThat(m.getKeyspace(keyspace).getTable("table1")).isNotNull();
+        for (SchemaChangeTestListener listener : listeners) {
+            ArgumentCaptor<TableMetadata> added = ArgumentCaptor.forClass(TableMetadata.class);
+            verify(listener, atLeastOnce()).onTableAdded(added.capture());
```

Comment:
This should be `times(1)` but for a strange reason, when running with Maven this yields unexpected numbers (running with IntelliJ is fine though).